lib/msf/core/exploit/http/client.rb

```diff
@@ -491,7 +491,9 @@ def full_uri(custom_uri = nil)
 
     uri = normalize_uri(custom_uri || target_uri.to_s)
 
-    if Rex::Socket.is_ipv6?(rhost)
+    if vhost_uri && datastore['VHOST']
```

Comment:
The keyword arg was deliberately named `vhost_uri` to avoid shadowing the `vhost` method in this file.

Reply:
Also, since we're checking `VHOST` specifically, `vhost` (fallback on `rhost`) is not used.